### PR TITLE
Adding ES6 Set as displayable body

### PR DESCRIPTION
**Pull request by Yomguithereal**
Adding ES6 set as displayable body. I chose to show it as `#{1, 2, 3}` but we should discuss this point.

### CODE REVIEW

File: example/datatypes.card.js [hunk 1]
```diff
@@ -72,6 +72,11 @@ var ListToggle = React.createClass({
   }
 });
 
+devcard('Set',
```
Comment (glenjamin) [inline example/datatypes.card.js]: Does babel polyfill this, or will it break on older browsers?

Comment (Yomguithereal) [inline example/datatypes.card.js]: I think you need to add the babel runtime for older browsers yes. Don't remember if webpack doesn't inject some kind of polyfill for you however.

Comment (glenjamin) [inline example/datatypes.card.js]: Is https://github.com/glenjamin/devcards-js/blob/master/example/.babelrc enough to guarantee this'll be fine?

File: lib/components/Body.js [hunk 2]
```diff
@@ -50,6 +56,15 @@ var Body = React.createClass({
     if (isDate(body)) {
       return $(Data, null, body.toString());
     }
+    if (isSet(body)) {
+      var contents = Array.from(body)
```
Comment (glenjamin) [inline lib/components/Body.js]: Devcards core aims to be ES5 only, so you can't rely on `Array.from`.

You can use https://www.npmjs.com/package/array.from though.

File: lib/components/Body.js [hunk 3]
```diff
@@ -50,6 +56,15 @@ var Body = React.createClass({
...
+        .map(function(item) {
```
Comment (glenjamin) [inline lib/components/Body.js]: I think it's worth extracting the `value -> string` function out of the component, so it can then be recursive.

**Comment by glenjamin** [comment]
Suggestion for textual representation:

```js
Set(a, b, c)
```

File: lib/components/Body.js [hunk 4]
```diff
@@ -28,6 +28,12 @@ function isRegex(x) {
   return !!(x.exec && x.test && x.toString);
 }
 
+function isSet(x) {
```
Comment (glenjamin) [inline lib/components/Body.js]: Your question earlier made me think about this a bit more, perhaps this is a better approach:

https://github.com/glenjamin/devcards-js/issues/9#issuecomment-180021654

File: lib/components/Body.js (hunk 2 above)
Comment (Yomguithereal) [inline lib/components/Body.js]: Don't need to use a package for that. Besides, if we want to be ES5 only, we shouldn't use Function.name (as in your tests on `x.constructor.name`).

Comment (Yomguithereal) [inline lib/components/Body.js]: But evolutions on #8 should handle the issue.

Comment (glenjamin) [inline lib/components/Body.js]: `function.name` has worked in browsers for ages - but I've also replaced those checks with node modules now anyway.

`Array.from` is ES6, and not well supported (same as `Object.assign`), so I'd rather use the package.

Comment (Yomguithereal) [inline lib/components/Body.js]: What I meant with `Don't need to use a package for that.` is that we can iterate on the Set very simply to create an Array from it.

Comment (glenjamin) [inline lib/components/Body.js]: Oh I see, the only examples I could find involved `for..of`, so I figured a ponyfill would be easier.

Comment (Yomguithereal) [inline lib/components/Body.js]: There are `Set.forEach` and `Set.values()`.

**Comment by Yomguithereal** [comment]
@glenjamin can I edit my PR safely or should I take the latest commits of the repo into account?

**Comment by glenjamin** [comment]
Looks like you'll need to rebase or merge since those other changes I made have minor conflicts

**Comment by Yomguithereal** [comment]
Ok. No problem. I'll try to edit the PR in the afternoon.

**Comment by glenjamin** [comment]
Closing, as `inspect-x` handles this.

**Pull request closed without merging**